src/Symfony/Component/OptionsResolver/Tests/OptionsResolverTest.php

```diff
@@ -778,6 +778,33 @@ public function testSetAllowedTypesFailsIfUnknownOption()
         $this->resolver->setAllowedTypes('foo', 'string');
     }
 
+    public function testResolveTypedWithUnion()
+    {
+        $this->resolver->setDefined('foo');
+        $this->resolver->setAllowedTypes('foo', 'string|int');
```

Comment:
it's the same as `->setAllowedTypes('foo', ['string', 'int']);` right? I like it